libraries/GCS_MAVLink/MissionItemProtocol.cpp

```diff
@@ -71,6 +71,7 @@ void MissionItemProtocol::handle_mission_count(
     if (packet.count > max_items()) {
         // FIXME: different items take up different storage space!
         send_mission_ack(_link, msg, MAV_MISSION_NO_SPACE);
+        gcs().send_text(MAV_SEVERITY_WARNING, "The max num of missions exceeded %d>%d", packet.count, max_items());
```

Comment:
```suggestion
        gcs().send_text(MAV_SEVERITY_WARNING, "Only %d mission items are supported", max_items());
```

This also doesn't really reflect if it's actually a mission, rally or fence, which makes the status text misleading/not terribly helpful... A better solution would probably be to add a mavlink command to query the maximum number that are supported so that a GCS can actually usefully query it.